### PR TITLE
Add typechecking for array argument in BufferAttribute

### PR DESCRIPTION
**Pull request by yn5**
Oversaw the required type for the argument and therefore spent some time figuring out why my regular `Array` wasn't working.

### CODE REVIEW

**Comment by mrdoob** [comment]
What were you passing?

**Comment by yn5** [comment]
A regular `Array` with integers. `[1, 2, 3]` for example.

File: src/core/BufferAttribute.js [hunk 1]
```diff
@@ -10,6 +10,20 @@ import { _Math } from '../math/Math';
 
 function BufferAttribute( array, itemSize, normalized ) {
 
+	var typeofArray = typeof array;
+
+	if (
+		(typeofArray !== UInt8Array) &&
```
Comment (mrdoob) [inline src/core/BufferAttribute.js]: Third should be `Uint8Array`.

**Comment by WestLangley** [comment]
Are not the docs sufficient here -- as long as they are correct?

three.js does not normally do type checking. Is the intention to change the policy and add type checking to the constructors?

**Comment by mrdoob** [comment]
We could also do...

```js
if ( array.buffer instanceof ArrayBuffer === false )
```

**Comment by yn5** [comment]
@mrdoob Thanks for your suggestions, I've amended the commit to reflect them. @WestLangley I agree that being still for discussion, in my opinion it would be a convenience

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**